sdk/nodejs/npm/npm_test.go

```diff
@@ -16,68 +16,44 @@ package npm
 
 import (
 	"context"
-	"os"
-	"path/filepath"
+	"fmt"
 	"testing"
 
-	pulumi_testing "github.com/pulumi/pulumi/sdk/v3/go/common/testing"
 	"github.com/stretchr/testify/assert"
 )
 
-func chdir(t *testing.T, dir string) {
-	cwd, err := os.Getwd()
-	assert.NoError(t, err)
-	assert.NoError(t, os.Chdir(dir)) // Set directory
-	t.Cleanup(func() {
-		assert.NoError(t, os.Chdir(cwd)) // Restore directory
-		restoredDir, err := os.Getwd()
-		assert.NoError(t, err)
-		assert.Equal(t, cwd, restoredDir)
-	})
-}
-
-//nolint:paralleltest // mutates environment variables, changes working directory
-func TestNPMInstall(t *testing.T) {
-	testInstall(t, "npm", false /*production*/)
-	testInstall(t, "npm", true /*production*/)
-}
-
-//nolint:paralleltest // mutates environment variables, changes working directory
-func TestYarnInstall(t *testing.T) {
-	t.Setenv("PULUMI_PREFER_YARN", "true")
-	testInstall(t, "yarn", false /*production*/)
-	testInstall(t, "yarn", true /*production*/)
-}
-
-func testInstall(t *testing.T, expectedBin string, production bool) {
-	// Skip during short test runs since this test involves downloading dependencies.
-	if testing.Short() {
-		t.Skip("Skipped in short test run")
+// This test checks that os.exec call for `npm install` is well constructed.
+func TestNPMInstallCmd(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		production   bool
+		expectedArgs []string
+	}{
+		{
+			production:   true,
+			expectedArgs: []string{"/bin/npm", "install", "--loglevel=error", "--production"},
+		}, {
+			production:   false,
+			expectedArgs: []string{"/bin/npm", "install", "--loglevel=error"},
+		},
 	}
 
-	// Create a new empty test directory and change the current working directory to it.
-	tempdir := t.TempDir()
-	chdir(t, tempdir)
-
-	// Create a package directory to install dependencies into.
-	pkgdir := filepath.Join(tempdir, "package")
-	assert.NoError(t, os.Mkdir(pkgdir, 0o700))
-
-	// Write out a minimal package.json file that has at least one dependency.
-	packageJSONFilename := filepath.Join(pkgdir, "package.json")
-	packageJSON := []byte(`{
-	    "name": "test-package",
-	    "dependencies": {
-	        "@pulumi/pulumi": "latest"
-	    }
-	}`)
-	assert.NoError(t, os.WriteFile(packageJSONFilename, packageJSON, 0o600))
-
-	// Install dependencies, passing nil for stdout and stderr, which connects
-	// them to the file descriptor for the null device (os.DevNull).
-	pulumi_testing.YarnInstallMutex.Lock()
-	defer pulumi_testing.YarnInstallMutex.Unlock()
-	bin, err := Install(context.Background(), pkgdir, production, nil, nil)
-	assert.NoError(t, err)
-	assert.Equal(t, expectedBin, bin)
+	pkgManager := &npmManager{
+		executable: "/bin/npm", // a fake path for testing.
```

Comment:
Let's make this `false` so that it never actually runs successfully.
(`/usr/bin/false` will always fail.)
```suggestion
		executable: "false", // a fake path for testing.
```